src/chain.js

```diff
@@ -2,14 +2,10 @@ const http = require('http');
 const { getBasic } = require('./utils/get_basic');
 
 /**
- * @param {http.ClientRequest} request
- * @param {net.Socket} source
- * @param {buffer.Buffer} head
- * @param {*} handlerOpts
- * @param {*} server
+ * @param {*} options
```

Comment:
would be good to document each of the options

Reply:
They will be removed when we switch to TypeScript. Won't have any effect then.

Reply:
They would be converted to an interface, not removed, right?

Reply:
yep, but the jsdocs won't be there since TypeScript will make them moot

Reply:
Can we JSDoc an interface? Aka have the comments explaining what's what on the interface itself? I mean, I have literally no idea what some of those are.

Reply:
sure, that is what i'd like to see in the end. we don't need to comment every single property, but some are definitely worth a few extra words.

```ts
interface FooBar {
  /** this is a foo */
  foo: number;
  /** this is a bar */
  bar: string;
}
```

those comments won't be presented via intellisense, but they are there when hovering over the propert

![image](https://user-images.githubusercontent.com/615580/133637668-d1397f8e-e235-45d3-8f3f-b7325bed0c11.png)

Reply:
Indeed, without the comments, it will be very hard for anyone to maintain this code in the future

Reply:
@jancurn With due respect, the previous implementation was very hard to maintain in contrast to this one. TypeScript will make it more understandable as well. By knowing the Node.js API and reading the code, the meaning of the properties is very clear. I'm not saying that we mustn't doc this. Just doing so won't help that much IMHO. Maybe this is just me and needs to be documented 100%.

Reply:
I have had a discussion with @mnmkng and decided to add comments anyway. Here's the golden rule I very liked:

> Always leave the campground cleaner than you found it

:D